UNIXFS.md

```diff
@@ -86,3 +128,72 @@ If there is only a single chunk, no intermediate unixfs file nodes are created,
 ## Exporting
 
 To read the file data out of the unixfs graph, perform an in order traversal, emitting the data contained in each of the leaves.
+
+## Design decision rationale
+
+### Metadata
+
+Metadata support in UnixFSv1.5 has been expanded to increase the number of possible use cases.  These include rsync and filesystem based package managers.
+
+Several metadata systems were evaluated:
+
+#### Separate Metadata node
+
+In this scheme, the existing `Metadata` message is expanded to include additional metadata types (`mtime`, `mode`, etc).  It then contains links to the actual file data but never the file data itself.
+
+This was ultimately rejected for a number of reasons:
+
+1. You would always need to retrieve an additional node to access file data which limits the kind of optimizations that are possible.
+
+	For example many files are under the 256KiB block size limit, so we tend to inline them into the describing UnixFS `File` node.  This would not be possible with an intermediate `Metadata` node.
+
+1. The `File` node already contains some metadata (e.g. the file size) so you'd end up with metadata stored in multiple places
+
```

Comment:
also note: this would have had forwards-compatibility issues that would have been annoying.